### PR TITLE
Create a dedicated module for api

### PR DESCRIPTION
**Pull request by mvladev**
**What this PR does / why we need it**:

Reduces dependency hell

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

I've split the PR into 3 separate commits to make the review easier.

**Release note**:

```noteworthy operator
A new module `github.com/gardener/etcd-druid/api` can be used to get the API definitions.
```

```action developer
Switch to `github.com/gardener/etcd-druid-api` if you vendor only the API of etcd-druid.
```

### CODE REVIEW

**Comment by amshuman-kr** [comment]
@mvladev I like it. It makes sense. This should ideally be the general pattern for most controllers that have resource definitions, right?

File: .ci/pipeline_definitions [hunk 1]
```diff
@@ -42,6 +42,9 @@ etcd-druid:
           preprocess: 'finalize'
         release:
           nextversion: 'bump_minor'
+          git_tags:
+          - ref_template: 'refs/tags/{VERSION}'
+          - ref_template: 'refs/tags/api-{VERSION}'
```
Comment (mvladev) [inline .ci/pipeline_definitions:47]: @AndreasBurger can you have a look if this is correct?

I want to have the standard version tag and `api-{VERSION}` tag as well.

**Comment by mvladev** [comment]
> This should ideally be the general pattern for most controllers that have resource definitions, right?

Yes. This is the idea.

**Comment by amshuman-kr** [comment]
@mvladev Sorry for the delay in review. There is an upcoming release this week for etcd-backup-restore and etcd-druid and we want to pick this after that release.

**Comment by amshuman-kr** [comment]
@mvladev To avoid dealing with the conflicts in vendoring during rebase, I have cherry-picked that last two commits from this PR in a fresh PR https://github.com/gardener/etcd-druid/pull/169. Shall I close this PR in favour of that?

**Pull request closed without merging**